lib/rules/one-var.js

```diff
@@ -62,20 +120,46 @@ module.exports = function(context) {
         "FunctionDeclaration": startFunction,
         "FunctionExpression": startFunction,
         "ArrowFunctionExpression": startFunction,
+        "BlockStatement": startBlock,
```

Comment:
hmmm… sorry guys: one more thing. I tried redeclaring a variable with `let` inside a switch statement and got a parse error from babel. Are we sure that switch statements are valid?

https://babeljs.io/repl/#?experimental=false&evaluate=true&loose=false&spec=false&code=let%20a%20%3D%20true%0Alet%20b%20%3D%20false%0A%0Aswitch%20(a)%20%7B%0A%20%20case%20true%3A%0A%20%20%20%20let%20b%20%3D%202%0A%20%20%20%20break%0A%20%20case%20false%3A%0A%20%20%20%20let%20c%20%3D%203%0A%20%20%20%20break%0A%7D